src/main/scala/firrtl/features/LetterCaseTransform.scala

```diff
@@ -0,0 +1,38 @@
+// See LICENSE for license details.
+
+package firrtl.features
+
+import firrtl.{analyses, Namespace, passes, Transform}
+import firrtl.options.Dependency
+import firrtl.stage.Forms
+import firrtl.transforms.ManipulateNames
+
+/** Parent of transforms that do change the letter case of names in a FIRRTL circuit */
+abstract class LetterCaseTransform extends ManipulateNames {
+  override def prerequisites = Seq(Dependency(passes.LowerTypes))
+  override def optionalPrerequisites = Seq.empty
+  override def optionalPrerequisiteOf = Forms.LowEmitters
+  override def invalidates(a: Transform) = a match {
+    case _: analyses.GetNamespace => true
+    case _                        => false
+  }
+
+  protected def newName: String => String
+
+  final def condition = _ => true
+
+  final def manipulate = (a: String, ns: Namespace) => newName(a) match {
+    case `a` => a
+    case b   => ns.newName(b)
+  }
+}
```

Comment:
This is a method that returns a new function every time it's called, is that what you want? I grant there maybe CSE in Scala but I'm not sure for this kind of thing

Reply:
Actually nevermind, I'm sure it's just a single function, the method is returning a function literal.

Reply:
Yeah, it might be more clear that it doesn't have meaningful space overhead as a `val`, but a pure parameterless def with a constant return value should be fine.